tests/flow/utils.py

```diff
@@ -0,0 +1,66 @@
+from RLTest import Env
+import multiprocessing as mp
+
+con = None
+
+def init_process_local_connection(env):
+    global con
+    con = env.getConnection()
+    return 1
+
+class Multiproc:
+
+    # create and init n_procs workers and their connections which will execute fn(args)
+    def __init__(self, env, n_procs):
+        # on macOS the spawn start method is now the default one since python 3.8.
+        # the spawn start methon fails when it stumbles upon objects that contain locks/fds
+        # For this reason need to forcefully set the context to fork.
+        # For more details see: https://bugs.python.org/issue33725
+        # or the python docs: https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
+        self.ctx = mp.get_context('fork')
+
+        # creating connetion on pool init for increasing the fn calls rate, also this way the env is not being pickled
+        # we are passing the env and not the connection so that a new connection will be created for each process.
+        self.pool = self.ctx.Pool(n_procs, initializer=init_process_local_connection, initargs=(env,))
+    
+    def __del__(self):
+        self.pool.terminate()
+
+    def __enter__(self):
+        self.pool._check_running()
```

Comment:
Will `_check_running` throw an exception incase we're no running ?
seems like you return `self` regardless of this call outcome.

Reply:
Yes it raises an excpetion:
    def _check_running(self):
        if self._state != RUN:
            raise ValueError("Pool not running")